src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultRazorLanguageServerCustomMessageTarget.cs

```diff
@@ -307,5 +318,35 @@ private static bool SupportsCSharpCodeActions(JToken token)
 
             return providesCodeActions && resolvesCodeActions;
         }
+
+        public override Task<JObject[]> UpdateConfigurationAsync(
```

Comment:
I'd add a huge comment above this stating how it's a polyfill for VS, I'd also copy and paste that comment onto the base class

Reply:
Might make sense to rename to align with the spec piece that it's polyfilling.

Also, given that `JObject[]` is the serialized form of this result I'd update the return type to just be `object` and rely on StreamJsonRpc to serialize on your behalf.

```suggestion
        public override Task<object[]> WorkspaceConfigurationAsync(
```